### PR TITLE
Wpcs

### PR DESCRIPTION
**Pull request by NicktheGeek**
adds WPCS, CircleCI for WPCS, and fixes for errors returned with phpcbf and phpcs.

@reaktivstudios/reviewers this is ready to review.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "Genesis Design Palette Pro - Export CSS",
+  "description": "Adds a button to Genesis Design Palette Pro to allow export of  raw CSS file",
```
Comment (jjeaton) [inline composer.json]: extra spaces before raw.

File: gppro-export-css.php [hunk 2]
```diff
@@ -1,223 +1,212 @@
...
-if( ! defined( 'GPXCS_BASE' ) ) {
-	define( 'GPXCS_BASE', plugin_basename(__FILE__) );
+/**
+ * Plugin Name: Genesis Design Palette Pro - Export CSS
+ * Plugin URI: https://genesisdesignpro.com/
+ * Description: Adds a button to export raw CSS file
+ * Author: Reaktiv Studios
+ * Version: 1.0.2
+ * Requires at least: 3.7
+ * Author URI: http://andrewnorcross.com
```
Comment (jjeaton) [inline gppro-export-css.php]: Let's update this while we're at it to https://reaktivstudios.com/

File: gppro-export-css.php [hunk 3]
```diff
@@ -1,223 +1,212 @@
...
+ *
+ * @package gppro-export-css
+ */
+
+/**
+ * Copyright 2014 Andrew Norcross
```
Comment (jjeaton) [inline gppro-export-css.php]: Copyright 2018 Reaktiv Studios, Inc.

File: gppro-export-css.php [hunk 4]
```diff
@@ -1,223 +1,212 @@
...
-			// set a default message
-			$message	= __( 'There was an error with your export. Please try again later.', 'gppro-export-css' );
+			// Set a default message.
+			$message = __( 'There was an error with your export. Please try again later.', 'gppro-export-css' );
 
-			// no parent class present
-			if ( $_REQUEST['reason'] == 'noclass' ) {
-				$message	= __( 'The main Genesis Design Palette Pro files are not present.', 'gppro-export-css' );
+			// No parent class present.
+			if ( 'noclass' === $_REQUEST['reason'] ) { // WPCS: csrf ok.
```
Comment (jjeaton) [inline gppro-export-css.php]: Do we have to use $_REQUEST here, could we changed all of these for $_GET?

File: gppro-export-css.php [hunk 5]
```diff
@@ -226,131 +215,103 @@ public function export_css_file() {
 		$output = get_theme_mod( 'dpp_styles' );
 
 		if ( empty( $output ) ) {
-			$failure	= menu_page_url( 'genesis-palette-pro', 0 ).'&section=build_settings&export-css=failure&reason=nofile';
+			$failure = menu_page_url( 'genesis-palette-pro', 0 ) . '&section=build_settings&export-css=failure&reason=nofile';
```
Comment (jjeaton) [inline gppro-export-css.php]: do we need to escape this URL?

File: gppro-export-css.php [hunk 6]
```diff
@@ -226,131 +215,103 @@ public function export_css_file() {
...
-		//* Prepare and send the export file to the browser
+		// Prepare and send the export file to the browser.
 		header( 'Content-Description: File Transfer' );
 		header( 'Cache-Control: public, must-revalidate' );
 		header( 'Pragma: hack' );
 		header( 'Content-type: text/css; charset=utf-8' );
 		header( 'Content-Disposition: attachment; filename="gppro-custom.css"' );
 		header( 'Content-Length: ' . mb_strlen( $output ) );
-		echo str_replace( '&quot;', '"', esc_html( $output ) );
+		echo str_replace( '&quot;', '"', esc_html( $output ) ); // WPCS: xss ok.
```
Comment (jjeaton) [inline gppro-export-css.php]: is `esc_html` the right thing to run on a CSS file?

**Review by jjeaton — CHANGES_REQUESTED** [review]

**Comment by NicktheGeek** [comment]
@jjeaton the requested changes have been applied. I did a couple of quick tests to make sure it's all working too.

**Review by jjeaton — APPROVED** [review]

**Pull request merged**